ray-operator/controllers/ray/rayjob_controller_scheduled_test.go

```diff
@@ -0,0 +1,222 @@
+		// In the last scheduled state the cluster should still exist since ShutdownAfterJobFinishes is False
+		ctx := context.Background()
+		namespace := "default"
+		cronSchedule := "*/1 * * * *"
+		rayJob := rayJobTemplate("rayjob-scheduled-no-deletion", namespace)
+		rayJob.Spec.Schedule = cronSchedule
+		rayJob.Spec.ShutdownAfterJobFinishes = false
+		rayCluster := &rayv1.RayCluster{}
+
+		It("Verify RayJob spec", func() {
+			Expect(rayJob.Spec.ShutdownAfterJobFinishes).To(BeFalse())
+			Expect(rayJob.Spec.Schedule).To(Not(BeEmpty()))
+		})
+
+		It("should create a RayJob object with the schedule", func() {
+			err := k8sClient.Create(ctx, rayJob)
+			Expect(err).NotTo(HaveOccurred(), "failed to create test scheduled RayJob resource")
+		})
+
+		// Since the test can be run at any time, the RayJob might immediately
+		// transition past "Scheduled" to "New" or "Initializing" as it triggers a run.
+		// We first check if we have already passed the "Scheduled" state this
+		// accommodates the race condition if we only checked "Scheduled" and avoiding test flakiness.
+		It("should have a JobDeploymentStatus reflecting its scheduled, new, or initializing state", func() {
+			getStatusFunc := getRayJobDeploymentStatus(ctx, rayJob)
+
+			currentStatus, err := getStatusFunc()
+			Expect(err).NotTo(HaveOccurred(), "Failed to get RayJob status initially")
+
+			if currentStatus == rayv1.JobDeploymentStatusInitializing || currentStatus == rayv1.JobDeploymentStatusNew {
+				Expect(currentStatus).To(Or(Equal(rayv1.JobDeploymentStatusInitializing), Equal(rayv1.JobDeploymentStatusNew)), "RayJob was already Initializing or New")
+				return
+			}
+			// If it's not Initializing, then it should be sheduled
+			Eventually(getStatusFunc, time.Second*5, time.Millisecond*500).Should(
+				Or(
+					Equal(rayv1.JobDeploymentStatusScheduled),
+				),
+				"JobDeploymentStatus should be Scheduled, New or Initializing within 5 seconds",
+			)
+		})
+
+		// The cron job runs every minute so it will take at most 1 minute to run
+		It("should transition to the Initializing", func() {
+			Eventually(
+				getRayJobDeploymentStatus(ctx, rayJob),
+				time.Second*60, time.Microsecond*500).Should(Equal(rayv1.JobDeploymentStatusInitializing),
+				"JobDeploymentStatus should be Initializing")
+		})
+
+		It("should create a raycluster object", func() {
+			Eventually(
+				getRayClusterNameForRayJob(ctx, rayJob),
+				time.Second*15, time.Millisecond*500).Should(Not(BeEmpty()))
+			Eventually(
+
+				getResourceFunc(ctx, common.RayJobRayClusterNamespacedName(rayJob), rayCluster),
+				time.Second*3, time.Millisecond*500).Should(Succeed())
+		})
+
+		// We are checking if the LastScheduleTime is correctly set
+		It("should have LastScheduleTime updated in its status", func() {
+			rayJobLookupKey := types.NamespacedName{Name: rayJob.Name, Namespace: rayJob.Namespace}
+			fetchedRayJob := &rayv1.RayJob{}
+
+			var lastScheduleTime *time.Time
+			Eventually(func() bool {
```

Comment:
This Eventually and the two Expect blocks below can be a single Eventually block. See this as an example https://github.com/ray-project/kuberay/blob/master/ray-operator/controllers/ray/rayjob_controller_test.go#L442-L444